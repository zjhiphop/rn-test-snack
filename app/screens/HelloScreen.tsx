import { observer } from "mobx-react-lite"
import React, { FC } from "react"
import { ViewStyle } from "react-native"
import { Screen, Text } from "../components"
import { AppStackScreenProps } from "../navigators"
// import { useNavigation } from "@react-navigation/native"
// import { useStores } from "../models"

// STOP! READ ME FIRST!
// To fix the TS error below, you'll need to add the following things in your navigation config:
// - Add `Hello: undefined` to AppStackParamList
// - Import your screen, and add it to the stack:
//     `<Stack.Screen name="Hello" component={HelloScreen} />`
// Hint: Look for the üî•!

// REMOVE ME! ‚¨áÔ∏è This TS ignore will not be necessary after you've added the correct navigator param type
// @ts-ignore
export const HelloScreenComponent = observer(function HelloScreen() {
  // Pull in one of our MST stores
  // const { someStore, anotherStore } = useStores()

  // Pull in navigation via hook
  // const navigation = useNavigation()
  return <Text text="Hello" />
})

interface HelloScreenProps extends AppStackScreenProps<"Hello"> {}

export const HelloScreen: FC<HelloScreenProps> = observer(function HelloScreen() {
  return (
    <Screen style={$root} preset="scroll">
      <HelloScreenComponent></HelloScreenComponent>
    </Screen>
  )
})

const $root: ViewStyle = {
  flex: 1,
}
