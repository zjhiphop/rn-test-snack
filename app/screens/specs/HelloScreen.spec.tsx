// https://reactnativetesting.io/component/testing/

import { render } from "@testing-library/react-native"
import React from "react"
import renderer from "react-test-renderer"
import { HelloScreenComponent } from "../HelloScreen"

describe("Hello", () => {
  it("renders HelloScreen component", () => {
    const screen = render(<HelloScreenComponent></HelloScreenComponent>)
    expect(screen.getByText("Hello")).toBeTruthy()
  })

  it("Renders Snapshot correctly", () => {
    const tree = renderer.create(<HelloScreenComponent></HelloScreenComponent>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
