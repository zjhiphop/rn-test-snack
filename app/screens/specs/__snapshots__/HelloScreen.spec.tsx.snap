// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Renders Snapshot correctly 1`] = `
<Text
  style={
    Array [
      Object {},
      Array [
        Object {
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "fontFamily": "spaceGroteskRegular",
        },
        Object {
          "color": "#191015",
        },
      ],
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Hello
</Text>
`;

exports[`Hello Renders screen Test HelloScreen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
